Dev/vote content

huan5678 opened the pull request.

huan5678 committed:
GIT 提交信息：

功能：精煉認證流程並改善 UI 組件

- 在 `App.vue` 中精煉檢查非空令牌的邏輯，明確檢查空字符串、`null` 和 `undefined`
- 將 `checkUser` 和 `authCheck` 合併為一個函數，明確認證檢查與授權邏輯
- 用 URLSearchParams 替代 `vue-router` 的 useRoute 功能來提取 URL 查詢參數
- 在 `ComponentFooter.vue` 中實作 `showModal` 方法和聯絡表單模態組件
- 更新頁腳樣式以保持一致性並包括聯絡模態組件
- 調整 `ContactFormModal.vue` 中模態的樣式以改善 UI 展示
- 重新排列 `NavbarEl.vue` 項目並改進響應式設計行為
- 修改訊息存儲以在 `message.js` 中返回確認結果
- 在 `PollView.vue` 中實作刪除投票前的條件確認對話框
- 更新 `PollView.vue` 中顯示投票詳情的事件處理
- 重構 `VoteContent.vue` 以響應式處理會員屬性，改善表單互動和錯誤處理
- 從 `VoteContent.vue` 的 Swiper 組件中移除 navigation 屬性

Signed-off-by: 黃士桓 <huan5678@gmail.com>

huan5678 committed:
GIT 提交訊息：

feat: 增強評論系統與用戶界面互動

- 新增 `CommentComponent.vue` 以便管理評論功能，如發布新評論、處理評論更新、以及刪除評論。
- 實現新的 `CommentEditComponent.vue` 用於編輯評論，包含過渡效果和其他用戶界面互動。
- 從代碼庫中移除 `HelloWorld.vue` 組件。
- 更新 `CommentModal.vue`，使得模態窗口關閉時自動恢復溢出屬性。
- 引入 `turnDateTime` 實用函數，以友善的格式顯示時間差異，如“幾分鐘前”或“幾天前”。
- 調整 `PollView.vue` 以反映新的評論管理組件以及投票進度計算等變更。
- 修改在 `VoteView.vue` 刪除投票時的確認訊息文本。
- 其他雜項更改包括移除控制檯日誌以及為清晰度和更好的錯誤處理進行的輕微重構。

Signed-off-by: 黃士桓 <huan5678@gmail.com>

huan5678 committed:
GIT 提交信息：

重構：為了清晰度和用戶體驗重構評論系統

- 移除 `CommentComponent.vue` 中未使用的 `onBeforeMount` 生命週期鉤子及相關的 `showReplyMenu` 和 `showReply` 引用。
- 簡化 `handleDeleteComment` 函數，去除狀態參數。
- 簡化並重命名 `sendComment` 函數，處理回復和更新，無需分開的案例。
- 通過移除冗餘代碼並合併相關案例，整合並簡化 `handleComment`。
- 移除過時的 `clickOutsideDropdown` 函數，支持新的 `CommentContent.vue` 組件處理。
- 引入新的 `CommentContent.vue` 組件，用於處理評論顯示邏輯。
- 更新 `CommentEditComponent.vue`，改進編輯狀態處理和用戶介面反饋。
- 允許 `CommentEditComponent.vue` 處理子回復的用戶介面邏輯。

Signed-off-by: huan5678 <huan5678@gmail.com>

huan5678 committed:
GIT 提交訊息：

功能: 重構 CommentView API 交互及清理

- 從 `CommentView.vue` 移除未使用的導入 `useRoute`
- 修改 `getComments` 函式中的 API URL，以包含尾部斜槓
- 將 `getComments` 中的過濾邏輯從 `.filter()` 改為 `.reduce()` 以進行去重
- 更新錯誤處理，當回應資料不可用時提供一個後備的錯誤訊息
- 在模板中取消註釋過濾按鈕部分，以實現功能性
- 移除已註釋的表格和討論參與者數量不必要的列

Signed-off-by: huan5678 <huan5678@gmail.com>

huan5678 committed:
[fix]: remove console log

eee317 commented:
我留言之後，點選編輯想要更改留言
結果本來編輯的留言變成新留言XD
原本想要把 coool 改成 nice
<img width="589" alt="image" src="https://github.com/sp1daTomm/SelectWave/assets/38075054/19c9055a-87ab-49f2-98dd-ac700940fb08">

eee317 commented:
我是這篇文章的發起人，自己開的留言串，卻無法刪除自己的留言（會彈窗提示我沒有權限刪除）
也想問問，怎麼樣的權限才能刪除呢? :)

huan5678 committed:
[fix]: animation

eee317 commented:
有發現我在這版沒有辦法投票

huan5678 committed:
GIT 提交訊息:

功能：改善評論系統的用戶體驗及邏輯

- 在 `CommentComponent` 中添加一個 `stateType` 參考以管理評論的狀態，包括目標和類型。
- 修改 `CommentContent` 組件，讓它可以接收 `memberId` 屬性，並利用此來條件性地顯示選項下拉菜單。
- 修訂 `CommentEditComponent` 以包括 `originalTarget` 屬性，並在開始編輯時發出含有目標和動作類型的 `state` 事件。
- 調整 `CommentEditComponent` 中 placeholder 屬性的條件，以簡化邏輯。
- 引進樣式變更到 `CommentComponent` 中的回覆部分，增加它們之間的間距。
- 允許提交狀態通過 `stateType` 內容來決定，在 `CommentComponent` 中區分創建或更新動作。
- 在 `VoteContent.vue` 中啟用對已登入用戶的 '投票' 按鈕。

Signed-off-by: 黃士桓 <huan5678@gmail.com>
